### PR TITLE
MXRT600 update core speed

### PR DESCRIPTION
**Pull request by mmahadevan108**
The MXRT600 power driver is available in source format. PR for this is https://github.com/zephyrproject-rtos/hal_nxp/pull/67
We can now increase the core speed.

### CODE REVIEW

File: boards/arm/mimxrt685_evk/mimxrt685_evk_cm33_defconfig [hunk 1]
```diff
@@ -14,8 +14,7 @@ CONFIG_UART_INTERRUPT_DRIVEN=y
 CONFIG_CORTEX_M_SYSTICK=y
 CONFIG_GPIO=y
 CONFIG_PINMUX=y
-CONFIG_SYS_CLOCK_HW_CYCLES_PER_SEC=48000000
-
+CONFIG_SYS_CLOCK_HW_CYCLES_PER_SEC=250105263
```
Comment (MaureenHelm) [inline boards/arm/mimxrt685_evk/mimxrt685_evk_cm33_defconfig:17]: This is a strange number, is it correct?

**Review by MaureenHelm — CHANGES_REQUESTED** [review]

**Comment by MaureenHelm** [comment]
Labeled DNM until the nxp_hal dependency is merged

File: boards/arm/mimxrt685_evk/board.h [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef __INC_BOARD_H
+#define __INC_BOARD_H
+
+/*!< Board System oscillator settling time in us */
+#define BOARD_SYSOSC_SETTLING_US 260U
+/*!< Board xtal32K frequency in Hz */
+#define BOARD_XTAL32K_CLK_HZ 32768U
```
Comment (agansari) [inline boards/arm/mimxrt685_evk/board.h:12]: i don't see where this definition is used. If it's not used, can we remove it?

File: boards/arm/mimxrt685_evk/mimxrt685_evk_cm33_defconfig (hunk 1 above)
Comment (agansari) [reply boards/arm/mimxrt685_evk/mimxrt685_evk_cm33_defconfig:17]: checked the sdk code, the value is correct

**Review by agansari — COMMENTED** [review]
Mostly okay, please check the comments below.

File: soc/arm/nxp_imx/rt6xx/CMakeLists.txt [hunk 3]
```diff
@@ -11,6 +11,8 @@ zephyr_sources(
 zephyr_library_include_directories(
   ${ZEPHYR_BASE}/kernel/include
   ${ZEPHYR_BASE}/arch/${ARCH}/include
+  ${ZEPHYR_BASE}/boards/${ARCH}/${BOARD}
+  ${ZEPHYR_BASE}/boards/${ARCH}/mimxrt685_evk
```
[the remark this answers is not in the document]
Comment (mmahadevan108) [reply soc/arm/nxp_imx/rt6xx/CMakeLists.txt:15]: How do we handle board specific defines?

File: boards/arm/mimxrt685_evk/mimxrt685_evk_cm33_defconfig (hunk 1 above)
Comment (mmahadevan108) [reply boards/arm/mimxrt685_evk/mimxrt685_evk_cm33_defconfig:17]: The value is derived from the main PLL PFD.
24 (SYSOSC) * 22 (SYSPLL) * 18 / 19 (PFD0) / 2 (CORE_DIV)

File: boards/arm/mimxrt685_evk/board.h (hunk 2 above)
Comment (mmahadevan108) [reply boards/arm/mimxrt685_evk/board.h:12]: https://github.com/NXPmicro/zephyr/blob/MXRT600_Update_Core_Speed/soc/arm/nxp_imx/rt6xx/soc.c#L123-L125

File: soc/arm/nxp_imx/rt6xx/CMakeLists.txt (hunk 3 above)
[the remark this answers is not in the document]
Comment (mmahadevan108) [reply soc/arm/nxp_imx/rt6xx/CMakeLists.txt]: I have updated to use config options for board level defines

File: west.yml [hunk 4]
```diff
@@ -102,7 +102,7 @@ manifest:
       revision: 1c4fdba512b268033a4cf926bddd323866c3261a
       path: tools/net-tools
     - name: hal_nxp
-      revision: 12a6084be5710ec39d1cc7626582a3c470db1459
+      revision: pull/67/head
```
Comment (MaureenHelm) [inline west.yml:105]: The hal PR has now been merged, so you can update this to the new SHA.

Comment (mmahadevan108) [reply west.yml]: Updated.

**Review by agansari — APPROVED** [review]

**Comment by mmahadevan108** [comment]
I am not clear why the Documentation GitHub is failing build

**Comment by utzig** [comment]
> I am not clear why the Documentation GitHub is failing build

You will have to rebase after #30002 is merged.

**Review by MaureenHelm — APPROVED** [review]

**Pull request merged**